features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/notifications/NotificationSettingsState.kt

```diff
@@ -46,4 +50,10 @@ data class NotificationSettingsState(
         val systemNotificationsEnabled: Boolean,
         val appNotificationsEnabled: Boolean,
     )
+
+    /**
+     * Whether the advanced settings should be shown.
+     * This is true if the current push distributor is in a failure state or if there are multiple push distributors available.
+     */
+    val showAdvancedSettings: Boolean = currentPushDistributor.isFailure() || availablePushDistributors.size > 1
```

Comment:
This should probably use a `get()` option since otherwise I think doing a `state.copy(availableDistributors = ...)` would change the distributors but not this value, and the current state of the `currentPushDistributor` won't be taken into account.

Reply:
`state.copy()` will create a new object and so `showAdvancedSettings` will be computed again. Or am I missing something?

Reply:
Oh, never mind 🤦 . I somehow forgot this is equivalent to:

```kotlin
val showAdvancedSettings: Boolean

init {
	showAdvancedSettings = ...
}
```